ee/server/service/devices.go

```diff
@@ -16,6 +18,43 @@ func (svc *Service) RequestEncryptionKeyRotation(ctx context.Context, hostID uin
 	return svc.ds.SetDiskEncryptionResetStatus(ctx, hostID, true)
 }
 
+type migrateMDMDevicePayload struct {
+	Timestamp time.Time `json:"timestamp"`
+	Host      struct {
+		ID             uint   `json:"id"`
+		UUID           string `json:"uuid"`
+		HardwareSerial string `json:"hardware_serial"`
+	} `json:"host"`
+}
+
+func (svc *Service) TriggerMigrateMDMDevice(ctx context.Context, host *fleet.Host) error {
+	ac, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return err
+	}
+	if !ac.MDM.EnabledAndConfigured {
+		return fleet.NewMDMNotConfiguredError()
+	}
+	if !ac.MDM.MacOSMigration.Enable {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "not enabled"))
```

Comment:
Nit but I don't think those are really bad gateway errors? It's a user-configuration issue, to me it's closer to a 4xx (client error), the endpoint shouldn't have been called? A bad gateway would've been if we did call a third-party URL and that server failed to handle the request.